go/pkg/bertyprotocol/store_metadata_index.go

```diff
@@ -167,12 +168,39 @@ func (m *metadataStoreIndex) handleGroupAddDeviceSecret(event proto.Message) err
 	}
 
 	if m.ownMemberDevice.device.Equals(senderPK) {
-		m.sentSecrets[string(e.DestMemberPK)] = struct{}{}
+		m.sentSecrets[string(e.DevicePK)+string(e.DestMemberPK)] = struct{}{}
+		if chanSentSecrets, ok := m.chanSentSecrets[string(e.DevicePK)+string(e.DestMemberPK)]; ok {
+			chanSentSecrets <- struct{}{}
+		}
 	}
 
 	return nil
 }
 
+func (m *metadataStoreIndex) watchSentSecrets(memberPK crypto.PubKey) (chan struct{}, error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	memberRaw, err := memberPK.Raw()
+	if err != nil {
+		return nil, errcode.ErrDeserialization.Wrap(err)
+	}
+	ownDeviceRaw, err := m.ownMemberDevice.device.Raw()
+	if err != nil {
+		return nil, errcode.ErrDeserialization.Wrap(err)
+	}
+
+	m.logger.Debug("watchSentSecrets", zap.Any("ownDevice", ownDeviceRaw), zap.Any("memberPK", memberRaw))
+
+	channel, ok := m.chanSentSecrets[string(ownDeviceRaw)+string(memberRaw)]
+	if !ok {
+		channel = make(chan struct{}, 1)
```

Comment:
```suggestion
		channel = make(chan struct{}, 0)
```

to be tested, but would be better to have an unbuffered channel if possible

Reply:
In my case, I know the chan will be used only for one message. The buffer will prevent to block `handleGroupAddDeviceSecret` in `store_metadata_index.go` if the channel is not being read.

Reply:
> In my case, I know the chan will be used only for one message. The buffer will prevent to block handleGroupAddDeviceSecret in store_metadata_index.go if the channel is not being read.

even if this is completely possible in some case, by default, **"I know that it uses only one message"** and **"the channel maybe not read"** in the same phrase seems too much fragile

Can you double-check and confirm that you really need a buffer, and if so, that a buffer of 1 is a safe value?

Reply:
Resolved because this code will be removed, see #2312 (comment)